spec.md

```diff
@@ -639,6 +644,57 @@ If the [referrers API](#listing-referrers) returns a 404, the client MUST fallba
 The response SHOULD be an image index with the same content that would be expected from the referrers API.
 If the response to the [referrers API](#listing-referrers) is a 404, and the tag schema does not return a valid image index, the client SHOULD assume there are no referrers to the manifest.
 
+##### Pagination
+
+Paginated results can be retrieved by adding an `n` parameter to the request URL, declaring that the response SHOULD be limited to `n` results.
+
+Starting a paginated tags flow MAY begin as follows:
+
+```HTTP
+GET /v2/<name>/tags/list?n=<integer>
+```
+
+The above specifies that a tags response SHOULD be returned, from the start of the result set, ordered lexically, limiting the number of results to `n`.
+
+The response to such a request would look as follows:
+
+```HTTP
+200 OK
+Content-Type: application/json
+Link: <<url>?n=<n from the request>&last=<last tag value from previous response>>; rel="next"
+{
+    "name": "<name>",
+    "tags": [
+      "<tag>",
+      ...
+    ]
+}
+```
+
+To get the _next_ `n` entries, one can create a URL where the argument `last` has the value from `tags[len(tags)-1]`.
+If there are indeed more results, the URL for the next block is encoded in an [RFC5988](https://tools.ietf.org/html/rfc5988) `Link` header, as a "next" relation.
+If the header is not present, the client can assume that all results have been received.
+
+> __NOTE:__ In the request template above, note that the brackets are required. For example, if the url is `http://example.com/v2/hello-world/tags/list?n=20&last=b`, the value of the header would be `<http://example.com/v2/hello-world/tags/list?n=20&last=b>; rel="next"`.
```

Comment:
Maybe include also an example with a relative reference, such as the server returning just `/v2/hello-world/tags/list?n=20&last=b>; rel="next"`. Which I would expect to be more common, as the reply may be routed a few times until getting to the client and the ip/hostname/port the server is aware of may be different from the one the client should make the request to.